python/ray/serve/_private/deployment_state.py

```diff
@@ -1368,34 +1407,35 @@ def _set_target_state(self, target_info: DeploymentInfo) -> None:
             ):
                 ServeUsageTag.NUM_REPLICAS_LIGHTWEIGHT_UPDATED.record("True")
 
-        self._target_state = target_state
-        self._curr_status_info = DeploymentStatusInfo(
-            self.deployment_name, DeploymentStatus.UPDATING
-        )
-        self._replica_constructor_retry_counter = 0
-        self._backoff_time_s = 1
-
-        app_msg = f" in application '{self.app_name}'" if self.app_name else ""
-        logger.info(
-            f"Deploying new version of deployment {self.deployment_name}{app_msg}."
-        )
-
-    def _set_target_state_autoscaling(self, num_replicas: int) -> None:
-        """Update the target number of replicas based on an autoscaling decision.
-
-        This differs from _set_target_state because we are updating the
-        target number of replicas base on an autoscaling decision and
-        not a redeployment. This only changes the target num_replicas,
-        and doesn't change the current deployment status.
-        """
-
-        new_info = copy(self._target_state.info)
-        new_info.set_autoscaled_num_replicas(num_replicas)
-        new_info.version = self._target_state.version.code_version
-
-        target_state = DeploymentTargetState.from_deployment_info(new_info)
+        # Determine whether the updated target state is simply scaling
+        is_scale = target_state.compare_state_excluding_num_replicas(self._target_state)
+        # If deployment is not in the middle of updating, and this new
+        # target state only changes the number of replicas, then set
+        # status to be UPSCALING/DOWNSCALING
+        if self._target_state.info and is_scale:
+            if self._curr_status_info.status != DeploymentStatus.UPDATING:
+                new, old = (target_state.num_replicas, self._target_state.num_replicas)
+                # New replicas should never be equal to old replicas because both
+                # deploy() and autoscale() return early if the new number of replicas
+                # is the same, so _set_target_state should never get an exact copy of
+                # the existing target state.
+                assert new != old
+                scaling_decision = (
+                    DeploymentStatus.UPSCALING
+                    if new > old
+                    else DeploymentStatus.DOWNSCALING
+                )
+                self._curr_status_info.update(
+                    status=scaling_decision,
+                    status_trigger=status_trigger,
+                    message=f"{scaling_decision.capitalize()} from {old} to {new} replicas.",  # noqa
```

Comment:
[Nit] Why is there a `#noqa` here? Can you split the string into multiple lines with parentheses?